src/github.com/stellar/gateway/mocks/main.go

```diff
@@ -85,17 +85,52 @@ func (m *MockRepository) GetLastCursorValue() (cursor *string, err error) {
 
 func (m *MockRepository) GetAuthorizedTransactionByMemo(memo string) (*entities.AuthorizedTransaction, error) {
 	a := m.Called(memo)
-	return a.Get(0).(*entities.AuthorizedTransaction), a.Error(1)
+	if a.Get(0) == nil {
+		return nil, a.Error(1)
+	} else {
+		return a.Get(0).(*entities.AuthorizedTransaction), a.Error(1)
+	}
 }
 
 func (m *MockRepository) GetAllowedFiByDomain(domain string) (*entities.AllowedFi, error) {
 	a := m.Called(domain)
-	return a.Get(0).(*entities.AllowedFi), a.Error(1)
+	if a.Get(0) == nil {
+		return nil, a.Error(1)
+	} else {
+		return a.Get(0).(*entities.AllowedFi), a.Error(1)
```

Comment:
I highly recommend using golint and govet, it will really help to make your code more idomatic..  The if/else construction is less idiomatic than early return: 

```go
func (m *MockRepository) GetAllowedFiByDomain(domain string) (*entities.AllowedFi, error) {
    if a.Get(0) == nil {
        return nil, a.Error(1)
    }

    return a.Get(0).(*entities.AllowedFi), a.Error(1)
}
```